### ISSUE TITLE
Enabling GPU acceleration

### ISSUE THREAD

**Issue by pfebrer** [zerothi/sisl#518]
I found this project called `cupy`: https://cupy.dev/ that mimicks exactly the APIs of both `numpy` and `scipy`, but does the computations on GPU.

It seems to already have gained some popularity and traction, so perhaps it would be nice to keep it in mind if at some point we want to provide GPU acceleration for some algorithms in `sisl` :)

**Comment by zerothi** [comment]
Yes, I know of the project, however it probably requires that the vectors are pretty large before it makes sense to do anything with cupy, the overhead of communication is typically a bottleneck.

So it would require some testing and also that routines are built for both cpu and gpu usage (via flags or contexts or...) to make the most of sisl.

But great to have on the table!
